Command/DeleteOrphanIndexesCommand.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace EMS\CoreBundle\Command;
+
+use Elasticsearch\Client;
+use EMS\CoreBundle\Service\AliasService;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DeleteOrphanIndexesCommand extends EmsCommand
+{
+    /** @var LoggerInterface */
+    protected $logger;
+    /** @var Client */
+    protected $client;
+    /** @var AliasService */
+    protected $aliasService;
+
+    public function __construct(LoggerInterface $logger, Client $client, AliasService $aliasService)
+    {
+        $this->logger = $logger;
+        $this->client = $client;
+        $this->aliasService = $aliasService;
+        parent::__construct($logger, $client);
+    }
+
+    protected function configure()
+    {
+        $this->setName('ems:delete:orphans')
+            ->setDescription('Removes all orphan indexes');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->aliasService->build();
+        foreach ($this->aliasService->getOrphanIndexes() as $index) {
+            $this->client->indices()->delete([
```

Comment:
It might be useful to surround this by a try catch if it’s not possible to delete an index. With a log error in the catch. So even if it’s not possible to delete one orphan index the command can continue its cleaning

Reply:
And move the try catch and the delete into a private function deleteOrpheanIndex? To avoid nested code structure

Reply:
OK fixed it like this